tests/unit/policies.js

```diff
@@ -15,6 +347,20 @@ function prepareDummyRequest(headers = {}) {
 }
 
 describe('Policies: permission checks for S3 APIs', () => {
+    apiMatrix.forEach(api => {
+        if (api.name.length === 0) return;
+        const message = `should return ${api.expectedPermissions.join(', ')} in requestContextParams for ${api.name}` +
+            `${api.headers.length > 0 ? ` with headers ${api.headers.map(el => el[0]).join(', ')}` : ''}`;
+        it(message, () => {
+            const request = prepareDummyRequest(api.headers);
+            const requestContexts = prepareRequestContexts(api.name, request,
+                sourceBucket, sourceObject);
+            const listOfrequestedActions = requestContexts.map(rq => rq.getAction());
+            assert.strictEqual(listOfrequestedActions.length, api.expectedPermissions.length);
+            assert(api.expectedPermissions.every(value => listOfrequestedActions.includes(value)));
```

Comment:
```suggestion
            const requestedActions = requestContexts.map(rq => rq.getAction());
            assert.deepStrictEqual(requestedActions, api.expectedPermissions);
```

A deep strict equal will work here better ( in my opinion anyway ). This is cause using an every+includes will not catch if the list has more permissions then our expected. ( [a, b, c].every+includes[a, b, c, d] will be true ). This is why you also needed the length check. But because our permissions are in a deterministic order we can just skip these two checks and do a deep strict equal. Plus its less code. :smirk:

PS: requested actions is plural so we can skip `list of`